code/modules/projectiles/guns/energy/special.dm

```diff
@@ -305,15 +305,140 @@
+	bubble_icon = "swarmer"
+	light_color = "#89078E"
+	light_range = 3
+	light_power = 0.1
+	var/overloaded = FALSE
+	var/warned = FALSE
+	var/charging = FALSE
+
+/obj/item/gun/energy/toxgun/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/item/gun/energy/toxgun/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/obj/item/gun/energy/toxgun/process() //keep your finger on the trigger soldier
+	..()
+	if(overloaded)
+		cell.charge -= 0.5 //2.5 per second, 25 every 10 seconds
+		if(cell.charge <= 25 && !warned)
+			warned = TRUE
+			playsound(loc, 'sound/weapons/smg_empty_alarm.ogg', 75, 1)
+			atom_say("Caution, charge low. Forced discharge in under 10 seconds.")
+		if(cell.charge <= 5)
+			discharge()
+		if(ishuman(loc))
+			var/mob/living/carbon/human/user = loc
+			if(!user.is_in_active_hand(src))
+				discharge()
+		else
+			discharge()
+
+/obj/item/gun/energy/toxgun/attack_self(mob/living/user)
+	if(overloaded)
+		to_chat(user, "<span class='warning'>[src] is already overloaded!</span>")
+		return
+	if(cell.charge <= 140) //at least 6 seconds of charge time
+		to_chat(user, "<span class='warning'>[src] does not have enough charge to be overloaded.</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin to overload [src].</span>")
+	charging = TRUE
+	if(do_after(user, 25, target = src))
+		overloaded = TRUE
+		cell.charge -= 125
+		playsound(loc, 'sound/machines/terminal_prompt_confirm.ogg', 75, 1)
+		atom_say("Overloading successful.")
+		light_power = 4 //extra visual effect to make it more noticable to user and victims alike
+	charging = FALSE
+
+/obj/item/gun/energy/toxgun/proc/reset_overloaded()
+	light_power = 0.1 //setting it to zero doesn't update the light for some reason?
+	overloaded = FALSE
+	warned = FALSE
+
+/obj/item/gun/energy/toxgun/process_fire(atom/target, mob/living/user, message = TRUE, params, zone_override, bonus_spread = 0)
+	if(overloaded)
+		select_fire(user)
+	else if(charging)
+		return
+	return ..()
+
+/obj/item/gun/energy/toxgun/process_chamber()
+	if(overloaded)
+		select_fire(usr)
+		do_sparks(2, 1, src)
+		reset_overloaded()
+	..()
+	update_icon()
+
+/obj/item/gun/energy/toxgun/emp_act(severity)
+	..()
+	if(prob(100 / severity))
+		if(overloaded)
+			discharge()
+
+/obj/item/gun/energy/toxgun/proc/discharge() //25% of the time, plasma leak. Otherwise, shoot at a random mob / turf nearby. If no proper mob is found when mob is picked, fire at a turf instead
+	reset_overloaded()
+	do_sparks(2, 1, src)
+	update_icon()
+	if(prob(25))
+		visible_message("<span class='danger'>[src] vents heated plasma!</span>")
+		var/turf/simulated/T = get_turf(src)
+		if(istype(T))
+			T.atmos_spawn_air(LINDA_SPAWN_TOXINS|LINDA_SPAWN_20C,15)
+		return
+	if(prob(50))
+		var/found_someone = FALSE
+		var/list/mob_targets = list()
+		for(var/mob/living/M in range(get_turf(src), 7))
+			if(M in range(get_turf(src), 1)) // shooting at the person holding it does not work
+				continue
```

Comment:
instead, use the `orange` proc, this will omit the turf in the center.
better yet, I think `oview` would be better, doesn't make sense to be firing at someone who is behind a wall.

Reply:
Done